util/archutil/fixtures/exit.amd64.s

```diff
@@ -1,6 +1,50 @@
-	.global _start
-	.text
+                     // SSE3     SSSE3    CMPXCHNG16 SSE4.1    SSE4.2    POPCNT
+#define V2_FEATURES_CX (1 << 0 | 1 << 9 | 1 << 13  | 1 << 19 | 1 << 20 | 1 << 23)
+                         // LAHF/SAHF
+#define V2_EXT_FEATURES_CX (1 << 0)
+
+// 64-113 is user-defined exit codes
+// https://tldp.org/LDP/abs/html/exitcodes.html
+#define exitcode(x) $((x) + 64)
+
+  .global _start
+  .text
 _start:
-	mov $60, %rax
-	xor %rdi, %rdi
-	syscall
+  // highest basic calling parameter
```

Comment:
This can only determine between v1 and v2. This is enough as no emulator supports higher than v2 anyway and detecting the rest would just be unnecessary complexity. If this changes in the future this can be improved.

Go implementation used on native platforms detects v1-v4